backend/src/jippity.ts

```diff
@@ -0,0 +1,697 @@
+
+        if (isForceActionMessage(message)) {
+            // If there is a pending stimulusResolver, resolve it with this ForceActionMessage immediately
+            if (this.reactionQueue.length === 0 && this.stimulusResolver) {
+                log.debug(`Resolving stimulusResolver with force action message`);
+                const resolver = this.stimulusResolver;
+                this.stimulusResolver = null;
+                resolver.resolve({ type: "Message", message });
+                return;
+            }
+
+            // Remove all ContextMessages from reactionQueue and put them in contextBuffer
+            const remaining: ForceActionMessage[] = [];
+            for (const msg of this.reactionQueue) {
+                if (isContextMessage(msg)) {
+                    this.contextBuffer.push(msg);
+                } else {
+                    remaining.push(msg);
+                }
+            }
+            this.reactionQueue = remaining;
+            // Add the new ForceActionMessage
+            this.reactionQueue.push(message);
+            return;
+        }
+    }
+
+    private async waitForStimulus(): Promise<Stimulus> {
+        // Implementation that waits for either an idle timer or a message
+        if (this.reactionQueue.length > 0) {
+            log.debug(
+                `waitForStimulus: there are already ${this.reactionQueue.length} messages in reactionQueue`
+            );
+            const message = this.reactionQueue.shift()!;
+            log.debug(
+                `waitForStimulus: returning message stimulus from reactionQueue: ${message.command}`
+            );
+            return { type: "Message", message };
+        }
+        if (this.stimulusResolver) {
+            log.error(
+                "waitForStimulus: stimulusResolver already exists; please tell the maintainer about this"
+            );
+            throw new Error("stimulusResolver already exists");
+        }
+        this.stimulusResolver = Promise.withResolvers<Stimulus>();
+
+        let resolved = false;
+        const originalResolve = this.stimulusResolver.resolve;
+        const originalReject = this.stimulusResolver.reject;
+
+        // Wrap resolve/reject to ensure only one call
+        this.stimulusResolver.resolve = (value) => {
+            if (!resolved) {
+                resolved = true;
+                this.stimulusResolver = null;
+                originalResolve(value);
+            }
+        };
+        this.stimulusResolver.reject = (reason) => {
+            if (!resolved) {
+                resolved = true;
+                this.stimulusResolver = null;
+                originalReject(reason);
+            }
+        };
+
+        setTimeout(() => {
+            if (!resolved && this.stimulusResolver) {
+                log.debug("waitForStimulus: timeout reached, resolving with IdleTimer");
+                this.stimulusResolver.resolve({ type: "IdleTimer" });
+            }
+        }, 1000);
```

Comment:
Currently Jippity does not use the environment variable and the default value specified in the README for the action interval. This change uses the environment variable, applies the minimum of 1000ms, and uses the default value of 10000ms.

```suggestion
        }, Math.max(1000, parseInt(process.env.JIPPITY_INTERVAL_MS ?? "10000")));
```